test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs

```diff
@@ -19,22 +19,30 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
     public class HttpProtocolFeatureCollectionTests
     {
         private readonly TestHttp1Connection _http1Connection;
-        private readonly HttpConnectionContext _http1ConnectionContext;
+        private readonly HttpConnectionContext _httpConnectionContext;
         private readonly IFeatureCollection _collection;
+        private readonly IFeatureCollection _http2Collection;
 
         public HttpProtocolFeatureCollectionTests()
         {
-            _http1ConnectionContext = new HttpConnectionContext
+            var context = new Http2StreamContext
             {
                 ServiceContext = new TestServiceContext(),
                 ConnectionFeatures = new FeatureCollection(),
                 TimeoutControl = Mock.Of<ITimeoutControl>(),
                 Transport = Mock.Of<IDuplexPipe>(),
+                ServerPeerSettings = new Http2PeerSettings(),
+                ClientPeerSettings = new Http2PeerSettings(),
             };
 
-            _http1Connection = new TestHttp1Connection(_http1ConnectionContext);
+            _httpConnectionContext = context;
+            _http1Connection = new TestHttp1Connection(context);
```

Comment:
Http1Connection with a Http2StreamContext?

Reply:
Http2StreamContext derives from HttpConnectionContext and these tests need just enough properties set on the context to not null-ref in the constructor.